tests/queries/0_stateless/01744_alter_replace_partition_update.sql

```diff
@@ -0,0 +1,66 @@
+drop table if exists x;
+
+create table x (type Enum8('hourly' = 0, 'hourly_staging' = 1, 'daily' = 2, 'daily_staging' = 3), dt DateTime, i int)
+engine MergeTree partition by (type, if(toUInt8(type) < 2, dt, toStartOfDay(dt))) order by i;
```

Comment:
You removed the test for previous behavior where you modify the partition key column explicitly. Maybe return it back? Because even if I try to execute this:

    alter table x replace partition ('daily', '2021-03-28 00:00:00') update type = 'daily', dt = '2021-03-28 00:00:00' in partition ('hourly', '2021-03-28 07:00:00');

The previous row `('hourly', '2021-03-28 07:00:00')` is still there.

Reply:
Explained in previous comment.